README.md

```diff
@@ -25,113 +24,123 @@ Puppet Strings generates HTML documentation for Puppet extensions written in Pup
 
-### Generate module documentation
+By default, running `puppet strings` generates HTML documentation for a module into a `docs` directory within that module. To pass any options or arguments, use the `generate` action.
 
-To generate documentation for a Puppet module, run Strings from that module's directory. 
+Action   | Description   
+----------------|-------------------------
 
-1. Change directory into the module: `cd /modules/sample-module`.
-2. Run the command: `puppet strings`.
+`generate` | Generates documentation with the specified parameters, including format and output location.
+`server` | Serves documentation for all modules in the [module path](https://docs.puppet.com/puppet/latest/reference/dirs_modulepath.html) locally at `http://localhost:8808`.
 
-### Generate documentation for specific files
+### `puppet strings generate` command reference
 
-To generate documentation for specific files in a module, run the `puppet strings generate` subcommand and specify the files. 
+Usage: `puppet strings [generate] [--format <FORMAT>][--out <DESTINATION>] [<ARGUMENTS>]
+
+For example:
 
 ```
-puppet strings generate first.pp second.pp
+puppet strings generate --format markdown --out docs/info.md
 ```
 
-To generate documentation for specific directories, run the `puppet strings generate` command and specify the directories:
-
 ```
-$ puppet strings generate 'modules/foo/lib/**/*.rb' 'modules/foo/manifests/**/*.pp' 'modules/foo/functions/**/*.pp' ...
+puppet strings generate manifest1.pp manifest2.pp
 ```
 
-### Set additional options for document generation
+[--format <FORMAT>][--out <DESTINATION>] [<ARGUMENTS>]
 
-If you need to specify additional options when generating documentation, use the `puppet strings:generate` rake task. This command behaves exactly as `puppet strings generate`, but allows you to add the following parameters:
+Option   | Description   | Values      | Default
+----------------|:---------------:|:------------------:|-------------------------
+
+`--format` | Specifies a format for documentation. | Markdown, JSON    | If not specified, Strings outputs HTML documentation.
+`--out` | Specifies an output location for documentation | A valid directory location and filename.    | If not specified, outputs to default locations depending on format: HTML (`/docs/`), Markdown (main module directory), or JSON (stdout).
+Filenames or directory paths | Outputs documentation for only specified files or directories. | Markdown, JSON.    | If not specified, Strings outputs HTML documentation.
+
+
+## Setting additional options with Rake tasks
+
+You can use Puppet Strings Rake tasks to generate documentation with additional options or to make your generated docs available on [GitHub Pages](https://pages.github.com/).
+
+To specify additional options when generating documentation, use the `puppet strings:generate` rake task. This command behaves exactly as `puppet strings generate`, but allows you to add the following parameters:
```

Comment:
This is interesting that this is from the original documentation. As far as I can tell, you can totally specify most of these options, if not all, in the command line. Though I will say, yard_args on the command line is a little dodgy. I even added a warning to stderr to warn the user of this :)

Reply:
innnteresting... should we take this out? I'd rather just say you can specify the YARD options with the command, if that's true. Unless people are dying to use Rake for stuff?

Reply:
The help output is actually a little wonky but all those options are definitely available in the CLI

```shell
puppet strings generate --help


USAGE: puppet strings [--format OUTPUT_FORMAT]
[--out PATH]
[--markup FORMAT]
[--emit-json-stdout]
[--emit-json PATH]
[[search_pattern] ...]

Generate documentation from files.

OPTIONS:
  --render-as FORMAT             - The rendering format to use.
  --verbose                      - Whether to log verbosely.
  --debug                        - Whether to log debug information.
  --emit-json PATH               - DEPRECATED: Write JSON representation of the
                                   documentation to the given file.
  --emit-json-stdout             - DEPRECATED: Print JSON representation of the
                                   documentation to stdout.
  --format OUTPUT_FORMAT         - Designate output format, JSON or markdown.
  --markup FORMAT                - The markup format to use for docstring text
                                   (defaults to 'markdown').
  --out PATH                     - Write selected format to PATH. If no path is
                                   designated, strings prints to STDOUT.
```